okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -228,89 +173,36 @@ private boolean checkTokenCache(String tenant, HttpServerRequest req, String pat
-        rr = md.getProxyRoutingEntries();
-      } else if (id.equals(md.getId())) {
-        rr = md.getMultiRoutingEntries();
-      }
-      if (rr != null) {
-        Timer.Sample sampleLoopRoutingEntries = MetricsHelper.getTimerSample();
-        for (RoutingEntry re : rr) {
-          if (match(re, req)) {
-            ModuleInstance mi = new ModuleInstance(md, re, req.uri(), req.method(), true);
-
-            skipAuth = checkTokenCache(pc.getTenant(), req, re.getPathPattern(), mi);
-            mods.add(mi);
-            logger.debug("getModulesForRequest:  Added {} {} {} {} / {}", md.getId(),
-                re.getPathPattern(), re.getPath(), re.getPhase(), re.getLevel());
-            break;
-          }
-        }
-        MetricsHelper.recordCodeExecutionTime(sampleLoopRoutingEntries,
-            name + ".loopRoutingEntries");
+    for (ModuleInstance mi : mods) {
+      if (mi.isHandler()) {
+        RoutingEntry re = mi.getRoutingEntry();
+        skipAuth = checkTokenCache(pc.getTenant(), req, re.getPathPattern(), mi);
+        logger.debug("getModulesForRequest:  Added {} {} {} {} / {}",
+            mi.getModuleDescriptor().getId(),
+            re.getPathPattern(), re.getPath(), re.getPhase(), re.getLevel());
       }
-      rr = md.getFilterRoutingEntries();
-      Timer.Sample sampleLoopFilterEntries = MetricsHelper.getTimerSample();
-      for (RoutingEntry re : rr) {
-        if (match(re, req)) {
-          ModuleInstance mi = new ModuleInstance(md, re, req.uri(), req.method(), false);
-          mi.setAuthToken(req.headers().get(XOkapiHeaders.TOKEN));
-          mods.add(mi);
-          if (!resolveRedirects(pc, mods, re, enabledModules, "", req.uri())) {
-            MetricsHelper.recordCodeExecutionTime(sampleLoopFilterEntries,
-                name + ".loopFilterEntries");
-            MetricsHelper.recordCodeExecutionTime(sampleLoopEnabledModules,
-                name + ".loopEnabledModules");
-            MetricsHelper.recordCodeExecutionTime(sample, name);
-            return null;
-          }
-          logger.debug("getModulesForRequest:  Added {} {} {} {} / {}", md.getId(),
-              re.getPathPattern(), re.getPath(), re.getPhase(), re.getLevel());
-        }
+    }
+    for (ModuleInstance mi : mods) {
+      if (!mi.isHandler()) {
+        mi.setAuthToken(req.headers().get(XOkapiHeaders.TOKEN));
       }
-      MetricsHelper.recordCodeExecutionTime(sampleLoopFilterEntries, name + ".loopFilterEntries");
     }
-    MetricsHelper.recordCodeExecutionTime(sampleLoopEnabledModules, name + ".loopEnabledModules");
-
-    Timer.Sample sampleSortModuleInstances = MetricsHelper.getTimerSample();
-    Comparator<ModuleInstance> cmp = (ModuleInstance a, ModuleInstance b)
-        -> a.getRoutingEntry().getPhaseLevel().compareTo(b.getRoutingEntry().getPhaseLevel());
+    Comparator<ModuleInstance> cmp = Comparator.comparing((ModuleInstance a)
+        -> a.getRoutingEntry().getPhaseLevel());
```

Comment:
Wondering if we should extract this so it won't be executed every time?

Reply:
I don't see how we could save this sorting.. Having said that we're sorting - in most cases two modules: auth + handler module.. Suppose a few audit.. and we're up to 4.

Reply:
I mean define it just once somewhere and then use it every time.

Reply:
ok. done